test/e2e/multikueue/e2e_test.go

```diff
@@ -60,36 +83,157 @@ var _ = ginkgo.Describe("MultiKueue", func() {
+		gomega.Expect(k8sWorker1Client.Create(ctx, worker1Lq)).Should(gomega.Succeed())
+
+		worker2Flavor = utiltesting.MakeResourceFlavor("default").Obj()
+		gomega.Expect(k8sWorker2Client.Create(ctx, worker2Flavor)).Should(gomega.Succeed())
+
+		worker2Cq = utiltesting.MakeClusterQueue("q1").
+			ResourceGroup(
+				*utiltesting.MakeFlavorQuotas(worker2Flavor.Name).
+					Resource(corev1.ResourceCPU, "1").
+					Resource(corev1.ResourceMemory, "2G").
+					Obj(),
+			).
+			Obj()
+		gomega.Expect(k8sWorker2Client.Create(ctx, worker2Cq)).Should(gomega.Succeed())
+
+		worker2Lq = utiltesting.MakeLocalQueue(worker2Cq.Name, worker2Ns.Name).ClusterQueue(worker2Cq.Name).Obj()
+		gomega.Expect(k8sWorker2Client.Create(ctx, worker2Lq)).Should(gomega.Succeed())
 	})
 
 	ginkgo.AfterEach(func() {
 		gomega.Expect(util.DeleteNamespace(ctx, k8sManagerClient, managerNs)).To(gomega.Succeed())
 		gomega.Expect(util.DeleteNamespace(ctx, k8sWorker1Client, worker1Ns)).To(gomega.Succeed())
 		gomega.Expect(util.DeleteNamespace(ctx, k8sWorker2Client, worker2Ns)).To(gomega.Succeed())
-	})
-	ginkgo.When("Using multiple clusters", func() {
-		ginkgo.DescribeTable("Cluster kubeconfig propagation", func(c *client.Client, key string, ns **corev1.Namespace) {
-			readSecret := &corev1.Secret{}
-			gomega.Expect(k8sManagerClient.Get(ctx, types.NamespacedName{Namespace: "default", Name: "multikueue"}, readSecret)).To(gomega.Succeed())
-
-			cfg, err := clientcmd.RESTConfigFromKubeConfig(readSecret.Data[key])
-			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
-			remClient, err := client.New(cfg, client.Options{Scheme: k8sManagerClient.Scheme()})
-			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		util.ExpectClusterQueueToBeDeleted(ctx, k8sWorker1Client, worker1Cq, true)
+		util.ExpectResourceFlavorToBeDeleted(ctx, k8sWorker1Client, worker1Flavor, true)
 
-			// create an wl with the remote client
-			wl := utiltesting.MakeWorkload("wl", (*ns).Name).Obj()
-			gomega.Expect(remClient.Create(ctx, wl)).To(gomega.Succeed())
-
-			createdWl := &kueue.Workload{}
-			gomega.Expect((*c).Get(ctx, client.ObjectKeyFromObject(wl), createdWl)).To(gomega.Succeed())
-			gomega.Expect(createdWl.Spec).To(gomega.BeComparableTo(wl.Spec))
-
-		},
-			ginkgo.Entry("worker1", &k8sWorker1Client, "worker1.kubeconfig", &worker1Ns),
-			ginkgo.Entry("worker2", &k8sWorker2Client, "worker2.kubeconfig", &worker2Ns),
-		)
+		util.ExpectClusterQueueToBeDeleted(ctx, k8sManagerClient, managerCq, true)
+		util.ExpectResourceFlavorToBeDeleted(ctx, k8sManagerClient, managerFlavor, true)
+		util.ExpectAdmissionCheckToBeDeleted(ctx, k8sManagerClient, multiKueueAc, true)
 	})
 
+	ginkgo.When("Creating a multikueue admission check", func() {
+		ginkgo.It("Should run a job on worker if admitted", func() {
+			job := testingjob.MakeJob("job", managerNs.Name).
+				Queue(managerLq.Name).
+				Request("cpu", "2").
+				Request("memory", "1G").
+				Image("gcr.io/k8s-staging-perf-tests/sleep:v0.0.3", []string{"5s"}).
```

Comment:
Job running for 5s is long, especially if this strategy is to be used in more tests. Can we do better? One idea is to just send Delete to the running pods, as used in other tests. I'm ok keeping this for follow up, but add to description.

Reply:
At least for now, let's keep it running till the end , to have a graceful e2e scenario.

Reply:
done